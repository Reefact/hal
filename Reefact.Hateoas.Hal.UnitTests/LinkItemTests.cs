#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.Hateoas.Hal.UnitTests {

    [UseReporter(typeof(BeyondCompareReporter))]
    public class LinkItemTests {

        [Fact]
        public void LinkItemToStringTest() {
            // Setup
            LinkItem linkItem = new("/orders");
            linkItem.Name      = "ea";
            linkItem.Templated = true;
            linkItem.AddProperty("age", 10);
            // Exercise
            string hal = linkItem.ToString();
            // Verify
            Approvals.Verify(hal);
        }

    }

}