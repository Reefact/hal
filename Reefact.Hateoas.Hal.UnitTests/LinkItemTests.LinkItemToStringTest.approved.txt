{
  "name": "ea",
  "href": "/orders",
  "templated": true,
  "age": 10
}